### ISSUE TITLE
Documentation for NATS Bench data format

### ISSUE THREAD

**Issue by 610v4nn1** [D-X-Y/AutoDL-Projects#93]
I downloaded the archive with the evaluations of NATS Bench and I would like to get more information about the format of the data. In particular, the data linked in this page https://pypi.org/project/nats-bench/ at the 'archive' links.

I already have software for running my experiments, I would just like to convert the data format instead of integrating a new library, which may be more suitable for a different setting. 

Of course, I will cite the paper as suggested.

**Comment by D-X-Y** [comment]
@610v4nn1 
First of all, thanks for your interest.

If your project is based on Python, I would suggest you directly clone this repo (https://github.com/D-X-Y/NATS-Bench) and paste it into your project. The NATS-Bench has a minimal dependency: abc, bz2, pickle, numpy, and warning.

If you are interested in the data format, all the files are saved based on bz2/pickle, and you can use this function (https://github.com/D-X-Y/NATS-Bench/blob/main/nats_bench/api_utils.py#L89) it.
The loaded data in Python is usually a dict, where the key indicates the hyperparameters -- 12 or 90 or 200, and the value is all the information of a specific architecture trained with this hyperparameter. You can find more details at https://github.com/D-X-Y/NATS-Bench/blob/main/nats_bench/api_utils.py#L275.

Please let me know if you have any other questions!

**Comment by 610v4nn1** [comment]
Thanks for the answer @D-X-Y 

I managed to load and parse the dictionaries before, but I still have some problems interpreting the meaning of the keys.
I need to extract some simple informations:
* value of each hyperparameter (e.g., number of layers, different type of operations)
* value for each metric (e.g., accuracy) both on the validation set and on the test set when available. This also required to know at which epoch of training the measurement was made

I just want to get a simple CSV file with these info out of the complex dictionary format.

**Comment by 610v4nn1** [comment]
Just to give you an idea, my format is pretty much like this

`
id, hp_lr, hp_max_dropout, hp_max_units, ...., metric_validation_f1_micro, metric_validation_accuracy, metric_test_f1_micro
1, 0.1, 0.25, 128, ...., 0.1, 0.4, 0.12
`

**Comment by D-X-Y** [comment]
If so, a simple way could be to load NATS-Bench on your laptop and save the metrics that you are interested in to a CSV file; and finally, load this CSV in your software.
